pkg/stanza/fileconsumer/internal/fingerprint/fingerprint.go

```diff
@@ -18,50 +19,64 @@ const MinSize = 16 // bytes
 // Fingerprint is used to identify a file
 // A file's fingerprint is the first N bytes of the file
 type Fingerprint struct {
-	FirstBytes []byte
+	firstBytes []byte
+	HashBytes  []byte
+	NBytesUsed int
 }
 
 // New creates a new fingerprint from an open file
 func New(file *os.File, size int) (*Fingerprint, error) {
 	buf := make([]byte, size)
-
 	n, err := file.ReadAt(buf, 0)
 	if err != nil && !errors.Is(err, io.EOF) {
 		return nil, fmt.Errorf("reading fingerprint bytes: %w", err)
 	}
+	fBytes := buf[:n]
+
+	h := fnv.New128a()
+	h.Write(fBytes)
+	hash := h.Sum(nil)
 
 	fp := &Fingerprint{
-		FirstBytes: buf[:n],
+		firstBytes: fBytes,
+		HashBytes:  hash,
+		NBytesUsed: len(fBytes),
 	}
 
 	return fp, nil
 }
 
 // Copy creates a new copy of the fingerprint
 func (f Fingerprint) Copy() *Fingerprint {
-	buf := make([]byte, len(f.FirstBytes), cap(f.FirstBytes))
-	n := copy(buf, f.FirstBytes)
+	buf := make([]byte, len(f.firstBytes), cap(f.firstBytes))
+	n := copy(buf, f.firstBytes)
 	return &Fingerprint{
-		FirstBytes: buf[:n],
+		firstBytes: buf[:n],
+		HashBytes:  f.HashBytes,
+		NBytesUsed: f.NBytesUsed,
 	}
 }
 
+// UpdateFingerPrint will update fingerprint with new bytes
+func (f *Fingerprint) UpdateFingerPrint(offset int64, appendBytes []byte) {
```

Comment:
Noted elsewhere, offset is state which the reader managers and should not be a concern of the fingerprint.